src/v/kafka/server/group.cc

```diff
@@ -2816,16 +2816,16 @@ ss::future<error_code> group::remove() {
               _id);
         } else {
             vlog(
-              klog.error,
-              "Error occured replicating group {} delete records {} ({})",
+              klog.warn,
+              "Error occurred replicating group {} delete records {} ({})",
               _id,
               result.error().message(),
               result.error());
         }
     } catch (const std::exception& e) {
         vlog(
-          klog.error,
-          "Exception occured replicating group {} delete records {}",
+          klog.warn,
+          "Exception occurred replicating group {} delete records {}",
```

Comment:
it seems like we should log unknown exceptions at error level, and cherry pick out certain ones (like not_leader to log at warn level).

Reply:
we do not even need to cherry pick exceptions as all the failures are propagated as results. Reverted logging exceptions to `error` severity.